scripts/core/exp-native-enqueue.yml

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright (C) 2024 Intel Corporation
+#
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Unified Runtime Experimental API for enqueuing work through native APIs"
+ordinal: "100"
+--- #--------------------------------------------------------------------------
+type: enum
+extend: true
+typed_etors: true
+desc: "Extension enums to $x_device_info_t to support native enqueue."
+name: $x_device_info_t
+etors:
+    - name: NATIVE_ENQUEUE_SUPPORT_EXP
+      value: "0x2020"
+      desc: "[$x_bool_t] returns true if the device supports enqueueing of native work"
+--- #--------------------------------------------------------------------------
+type: enum
+extend: true
+desc: "Command Type experimental enumerations."
+name: $x_command_t
+etors:
+    - name: NATIVE_ENQUEUE_EXP
+      value: "0x2004"
+      desc: Event created by $xNativeEnqueueExp
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Native enqueue properties"
+name: $x_native_enqueue_flags_t
+etors:
+    - name: TBD
+      desc: "reserved for future use."
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Native enqueue properties"
+name: $x_native_enqueue_properties_t
+base: $x_base_properties_t
+members:
+    - type: $x_native_enqueue_flags_t
+      name: flags
+      desc: "[in] native enqueue flags"
+--- #--------------------------------------------------------------------------
+type: fptr_typedef
+desc: "Function enqueueing work through the native API to be executed immediately."
+name: $x_native_enqueue_function_t
+return: void
+params:
+    - type: void*
+      name: pUserData
+      desc: "[in][out] pointer to data to be passed to callback"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Immediately enqueue work through a native backend API"
+class: $x
+name: NativeEnqueueExp
```

Comment:
I consciously avoided the `urEnqueue` class since the interop func is being executed within the `urNativeEnqueueExp`. ie the interop func is not being enqueued, only the native work is being enqueued. A subtle difference, but not sure if this means it shouldn't be part of the `urEnqueue` class. Let me know what you think

Reply:
Its still part of the adapters queue scheduling is it not?

Reply:
fwiw, I vote for `urEnqueueNative`. Yes, the name might not be obvious at a glance, but the API itself I think makes it clear that we are invoking the callback immediately to enqueue work.

Reply:
So if I do:

```
urEnqueueMemBufferWrite(..., &e1);
urNativeEnqueueExp(..., func_that_reads_above_mem, e1, ...);
```
The adapter will make the dependency work, but the `func_that_reads_above_mem` is not enqueued itself. It is the enqueue-er rather than the enqueue-ee

Reply:
> The adapter will make the dependency work, but the `func_that_reads_above_mem` is executed immediately, not enqueued itself. It is the enqueue-er rather than the enqueue-ee

Okay, so I think using the `$xEnqueue` class. I think, to distance it from [clEnqueueNativeKernel](https://registry.khronos.org/OpenCL/sdk/3.0/docs/man/html/clEnqueueNativeKernel.html) and to remove the potential ambiguity of the `urEnqueueNative` name, I think it should be more specific.

For example, I think `urEnqueueNativeCommands` captures the essence of what you want to do.

Reply:
Have used this naming. Thanks